projects/dsp-ui/src/lib/viewer/views/property-view/property-view.component.ts

```diff
@@ -100,4 +96,30 @@ export class PropertyViewComponent implements OnInit, OnDestroy {
         this.addButtonIsVisible = true;
         this.propID = undefined;
     }
+
+    /**
+     * Given a resource property, check if an add button should be displayed under the property values
+     *
+     * @param prop the resource property
+     */
+    addValueIsAllowed(prop: PropertyInfoValues): boolean {
+        // temporary until CkEditor is integrated
+        const guiElement = (prop.propDef as ResourcePropertyDefinition).guiElement;
+        if (guiElement === 'http://api.knora.org/ontology/salsah-gui/v2#Richtext') {
+            return false;
+        }
+
+        const isAllowed = CardinalityUtil.createValueForPropertyAllowed(
+            prop.propDef.id, prop.values.length, this.parentResource.entityInfo.classes[this.parentResource.type]);
+
+        // check if:
+        // cardinality allows for a value to be added
+        // value component does not already have an add value form open
+        // user has write permissions
+        if (isAllowed && this.propID !== prop.propDef.id && this.addButtonIsVisible) {
```

Comment:
What does this do? `this.propID !== prop.propDef.id`

Reply:
"value component does not already have an add value form open"?

Reply:
`this.propID` is a string that is assigned a `prop.propDef.id` of the corresponding property when the add button is clicked. This logic is used to hide the add button when the add value form for that property is already open so that the user can't keep opening new add value forms. Without this logic, the user would always see the add button under the property if the cardinality allows for it.

Reply:
ok, I get it now :-)